### ISSUE TITLE
Don't use magic "behind the curtains"

### ISSUE THREAD

**Issue by kemo** [kohana/orm#104]
ORM shouldn't rely on magic methods natively but use the methods for setting / getting instead. Examples;
`$this->{$name} = $var;`
`$this->{$key}->values($values[$key], $column);`
`$this->$column = $values[$column];`
`$data[$var] = $this->{$var};`
`$this->{$property} = $value;`

This obviously doesn't apply to properties which aren't magic anyway

**Comment by kemo** [comment]
Any opinions?

**Comment by evanpurkhiser** [comment]
I'm a little bit confused by your example. Are you saying end-users should use `ORM::get` and `ORM::set` and we remove the magic methods that allow you to act on columns as if they were properties of the object?

``` php
// So instead of this
$my_model->some_column = 'value';

// It would be
$my_model->set('some_column', 'value');
```

Or are you saying internally we should never use magic methods? (In that case I think I would agree)

**Comment by kemo** [comment]
> Or are you saying internally we should never use magic methods?

@EvanPurkhiser exactly, sorry if I wasn't clear enough

**Comment by evanpurkhiser** [comment]
This sounds good to me then. I can't think of any reasons not to use the get/set methods. :+1:

**Comment by arteymix** [comment]
Yeah, even magic methods are mapped to `get` and `set`, so this would lighten the call stack. It would also be more flexible when dealing with uncommon column name containing spaces.
